orttraining/orttraining/core/framework/gradient_graph_builder.h

```diff
@@ -77,6 +77,13 @@ static std::unordered_map<std::string, std::unordered_set<size_t>>
 
 static std::unordered_set<std::string> INVERTIBLE_OPS{"LayerNormalization",
                                                       "Relu"};
+static std::unordered_set<int64_t> CAST_GRAD_ALLOWED_TYPES{
```

Comment:
Maybe a naive question. Curious is there any common definitions what types of data can be differentiable, which cannot?

Reply:
Hi @pengwa , I did not find any such existing definitions